controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -333,8 +350,42 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				},
 			),
 			builder.WithPredicates(prometheusPredicate),
-		).
-		Complete(r)
+		)
+	return ocsInitializationController.Complete(r)
+}
+
+func (r *OCSInitializationReconciler) ensureClusterClaimExists() error {
+	if !r.availableCrds[ClusterClaimCrdName] || len(r.clusters.GetStorageClusters()) < 1 {
+		return nil
+	}
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		r.Log.Error(err, "failed to get operator's namespace. retrying again")
+		return err
+	}
+
+	OdfInfoNamespacedName := types.NamespacedName{
+		Namespace: operatorNamespace,
+		Name:      storagecluster.OdfInfoConfigMapName,
+	}.String()
+
+	cc := &v1alpha1.ClusterClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: util.OdfInfoNamespacedNameClaimName,
+		},
+	}
+
+	_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, cc, func() error {
+		cc.Spec.Value = OdfInfoNamespacedName
+		return nil
+	})
```

Comment:
Who and how we are going to handle deletion of this cluster claim?

Reply:
In odf-info's `ensureDeleted`

Reply:
Decided on manual deletion for now